tests/test_teacher_views.py

```diff
@@ -8,18 +8,53 @@ def test_all_grades(client):
     # Login
     rv = login(client, 'teacher@stevenscollege.edu', 'qwerty')
     assert b'Logged in' in rv.data
-    # Make sure teacher can access
+    # Make sure teacher can not access
     assert client.get('/course/216/session/1/all_grades').status_code == 403
     # Log out of teacher who should not have access
     rv = logout(client)
     assert b'TSCT Portal Login' in rv.data
 
     rv = login(client, 'teacher2@stevenscollege.edu', 'PASSWORD')
     assert b'Logged in' in rv.data
-    # should be able  to access
+    # should be able to access
     assert client.get('/course/216/session/1/all_grades').status_code == 200
+    assert client.get('/course/180/session/1/all_grades').status_code == 403
+    assert client.get('/course/216/session/2/all_grades').status_code == 403
 
     response = client.get('/course/216/session/1/all_grades')
     # Making sure students grades are displaying
     assert b'bob phillp' in response.data
     assert b'<td>F</td>' in response.data
+    rv = logout(client)
+    assert b'TSCT Portal Login' in rv.data
+
+
+def test_assignment_grades(client):
+    # Make sure anonymous users cannot access
+    assert client.get('/course/180/session/2/assignments/1/grades').status_code == 302
+    # login
+    rv = login(client, 'teacher2@stevenscollege.edu', 'PASSWORD')
+    assert b'Logged in' in rv.data
+    # Make sure other teachers don't have access
+    assert client.get('/course/180/session/2/assignments/1/grades').status_code == 403
+    # Logout of other teacher
+    rv = logout(client)
```

Comment:
Make sure you test the coverage of the file. If the 403 statements do not boost coverage they are useless. If they do then disregard this comment.